src/mpid/ch4/src/ch4_comm.h

```diff
@@ -201,6 +203,10 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_Comm_create_hook(MPIR_Comm * comm)
         }
     }
 
+#if defined HAVE_LIBHCOLL
+    hcoll_comm_create(comm, NULL);
+#endif
+
```

Comment:
I think there are some valid reasons for doing the rank-map detection/creation prior to calling the NM/SHM hooks here. @yfguo is looking into if this is OK. I think @ddurnov is also interested in this change.